openmdao/core/system.py

```diff
@@ -468,7 +484,17 @@ def _get_root_vectors(self, vector_class, initial):
                 if not initial:
                     root_vectors[key][vec_name] = self._vectors[key][vec_name]._root_vector
                 else:
-                    root_vectors[key][vec_name] = vector_class(vec_name, type_, self)
+
+                    # Check for complex step to set vectors up appropriately.
+                    # If any subsystem needs complex step, then we need to allocate it everywhere.
+                    alloc_complex = 'cs' in self._approx_schemes
+                    for sub in self.system_iter(include_self=True, recurse=True):
+                        if alloc_complex:
+                            break
+                        alloc_complex = 'cs' in sub._approx_schemes
+
+                    root_vectors[key][vec_name] = vector_class(vec_name, type_, self,
```

Comment:
We would only ever want to declare set alloc_complex to True for the `vecname='nonlinear'` vector, right? (unless we are going to support complex step around linear solvers